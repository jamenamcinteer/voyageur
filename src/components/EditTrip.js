import React from "react";
import TripHeader from "./Navigation/TripHeader";
import TripForm from "./Forms/TripForm";
import { PageSection } from "./StyledComponents/Layout";
import { connect } from "react-redux";

const EditTrip = props => {
  const trip = props.trips.find(trip => trip._id === props.match.params.id);

  return (
    <React.Fragment>
      {trip && (
        <React.Fragment>
          <TripHeader
            title="Edit Trip"
            theme={props.theme}
            backTo={`/trip/${props.match.params.id}`}
            trip={trip}
            budgetItems={props.budgetItems}
            expenses={props.expenses}
            auth={props.auth}
          />
          <PageSection>
            <TripForm theme={props.theme} trip={trip} history={props.history} />
          </PageSection>
        </React.Fragment>
      )}
    </React.Fragment>
  );
};

const mapStateToProps = (state, props) => {
  return {
    trips: state.trips,
    budgetItems: state.budgetItems,
    expenses: state.expenses,
    auth: state.auth
  };
};

export default connect(mapStateToProps)(EditTrip);
