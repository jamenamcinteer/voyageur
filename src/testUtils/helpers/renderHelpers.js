import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

export function renderWithReduxRouterAndTheme(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{ui}</Router>
        </ThemeProvider>
      </Provider>
    ),
    store,
    history
  };
}
