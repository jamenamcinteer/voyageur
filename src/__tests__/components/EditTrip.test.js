import React from "react";
import { cleanup } from "@testing-library/react";
import { createStore } from "redux";
import "react-dates/initialize";
import EditTrip from "../../components/EditTrip";
import {
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
} from "../../testUtils/fixtures/mockStoreData";
import { renderWithReduxRouterAndTheme } from "../../testUtils/helpers/renderHelpers";

afterEach(cleanup);

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

const store = createStore(() => ({
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
}));

test("can render with redux with custom store", () => {
  const { asFragment } = renderWithReduxRouterAndTheme(
    <EditTrip match={{ params: { id: "1" } }} />,
    {
      store
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
