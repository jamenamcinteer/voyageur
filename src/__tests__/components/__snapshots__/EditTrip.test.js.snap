// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa hZqfbl"
  >
    <a
      href="/trip/1"
    >
      <svg
        aria-label="Go Back"
        style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
        version="1.1"
        viewBox="0 0 199.404 199.404"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <polygon
            points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
          />
        </g>
      </svg>
    </a>
    <h1
      class="sc-bwzfXH OVfcq"
    >
      Edit Trip
    </h1>
    <button
      class="sc-htpNat fqCEYg"
    >
      <img
        alt="Account Options"
        class="sc-bxivhb jwtarI"
        src="https://picsum.photos/id/947/45/45"
      />
    </button>
  </div>
  <div
    class="sc-gZMcBi irilpd"
  >
    <div
      class="sc-gqjmRU jdetvG"
    />
    <div
      class="sc-VigVT bjQgmE"
    >
      <h2
        class="sc-htoDjs eJUAaw"
      >
        Hawaii
      </h2>
      <div
        class="sc-jTzLTM hwwuHy"
      >
        <h3
          class="sc-dnqmqq jgPViO"
        >
          Apr 26, 1970 - Oct 1, 8307
        </h3>
        <h3
          class="sc-iwsKbI kGErDA"
        >
          $595 / $2000
        </h3>
      </div>
      <div
        class="sc-bZQynM bJqPLz"
      >
        <span
          class="sc-gzVnrw daRrCx"
          style="width: 29.731%; background-color: rgb(12, 124, 136);"
        />
      </div>
    </div>
  </div>
  <div
    class="sc-ckVGcZ jWQnQO"
  >
    <div
      class="sc-fjdhpX joeOdX"
    >
      <label
        class="sc-jzJRlG ddfqFX"
        for="1"
      >
        Destination
      </label>
      <input
        class="sc-cSHVUG gMdUiP"
        id="1"
        type="text"
        value="Hawaii"
      />
    </div>
    <div
      class="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          class="sc-kAzzGY jucvHf"
        >
          <div
            class="sc-dnqmqq jnbCEU"
          >
            <label
              class="sc-iwsKbI dtSDPi"
              for="1"
            >
              Destination
            </label>
            <input
              class="sc-gZMcBi kJGmWv"
              id="1"
              type="text"
              value="04/26/1970"
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-startDate"
            >
              Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div />
          <button
            class="sc-VigVT efIboi"
            style="padding: 10px 0px;"
          >
            <span
              class="sc-jTzLTM gVGdst"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            class="sc-kGXeez EPBCl"
          >
            <div>
              <button
                class="sc-VigVT efIboi"
                style="padding: 10px 0px;"
              >
                <span
                  class="sc-jTzLTM gVGdst"
                >
                  Delete
                </span>
              </button>
            </div>
            <div
              class="sc-kgoBCf fWbtNX"
            >
              <a
                backgroundcolor="transparent"
                border="0"
                buttontype="link"
                buttonwidth="auto"
                class="sc-fjdhpX cPSYhm"
                href="/"
                style="padding: 10px 0px;"
              >
                <span
                  class="sc-jzJRlG cphvgw"
                >
                  Cancel
                </span>
              </a>
              <button
                class="sc-VigVT fLcMOh"
              >
                <span
                  class="sc-jTzLTM cntDpR"
                  color="#fff"
                >
                  Save
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="sc-chPdSV cWCjRk"
      style="padding: 10px 0px;"
    >
      <span
        class="sc-kgoBCf jOukbh"
      >
        Change Photo
      </span>
    </button>
    <div
      class="sc-hMqMXs ZOCRt"
    >
      <button
        class="sc-chPdSV cWCjRk"
        style="padding: 10px 0px;"
      >
        <span
          class="sc-kgoBCf jOukbh"
        >
          Delete
        </span>
      </button>
      <div
        class="sc-eNQAEJ XEuOY"
      >
        <a
          backgroundcolor="transparent"
          border="0"
          buttontype="link"
          buttonwidth="auto"
          class="sc-kGXeez hPXLNi"
          href="/"
          style="padding: 10px 0px;"
        >
          <span
            class="sc-kpOJdX edYuyx"
          >
            Cancel
          </span>
        </a>
        <button
          class="sc-chPdSV lnDAMS"
        >
          <span
            class="sc-kgoBCf jHsowC"
            color="#fff"
          >
            Save
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
